app/src/openxr/cpp/OpenXRLayers.h

```diff
@@ -324,6 +324,7 @@ class OpenXRLayerSurface : public OpenXRLayerBase<T, U> {
     return this->swapchain->AndroidSurface();
   }
 
+  virtual ~OpenXRLayerSurface() {};
```

Comment:
Further investigate this a little bit, looks like the cause is that we shouldn't call destroySurface() for OpenXRLayerSurface here